docs/life-of-a-feature.md

```diff
@@ -0,0 +1,29 @@
+# Life of a Feature
+
+## Do's
+
+1. Understand the goal by answering "what's the problem we're solving?".
+1. Write technical design docs.
+    - Do some researches and provide background information and overview.
+    - Find a simple and sustainable solution. Technical decisions usually come along with trade-offs among several options. We need a simple solution for a difficult problem. The solution should also be scalable to future changes or growth if possible.
+    - Split a large design into smaller ones, e.g. backend, UI, UX, etc.
+    - ```(IMPORTANT)``` Define data modeling by designing database schema and API by following [API style guide](https://github.com/bytebase/bytebase/blob/main/docs/api-style-guide.md).
```

Comment:
I would list schema design as a separate item and link our version management doc

Reply:
Done.